core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackService.java

```diff
@@ -360,11 +360,17 @@ public StackResponse getByNameInOrgWithEntries(String name, Long organizationId,
             return transactionService.required(() -> {
                 Organization organization = organizationService.get(organizationId, user);
                 Stack stack = stackRepository.findByNameAndOrganizationIdWithLists(name, organization.getId());
+                if (stack == null) {
+                    throw new NotFoundException(String.format(STACK_NOT_FOUND_EXCEPTION_TXT, name));
+                }
                 StackResponse stackResponse = conversionService.convert(stack, StackResponse.class);
                 stackResponse = stackResponseDecorator.decorate(stackResponse, stack, entries);
                 return stackResponse;
             });
         } catch (TransactionExecutionException e) {
+            if (e.getCause() instanceof NotFoundException) {
```

Comment:
This is unnecessary, the ```TransactionRuntimeExecutionExceptionMapper``` does this.

Reply:
When I don't have this I got a message like this: 
```
ERROR: status code: 403, message: Stack not found found with name 'lma'
```

Reply:
And with this explicit throw here the response will be 401?

Reply:
Nope, 403

Reply:
An idea to do this is the following:

In ```BaseExceptionMapper``` create this method:
```
protected String getErrorMessageFromThrowable(Throwable e) {
    return getErrorMessage((E) e);
}
```

and in ```TransactionRuntimeExecutionExceptionMapper``` do this:
```
@Override
    protected String getErrorMessage(TransactionRuntimeExecutionException exception) {
        TransactionRuntimeExecutionException deepest = getDeepestTransactionException(exception);
        return exceptionMappers.stream()
                .filter(m -> m.getExceptionType().equals(deepest.getCause().getCause().getClass()))
                .map(baseExceptionMapper -> baseExceptionMapper.getErrorMessageFromThrowable(deepest.getCause().getCause()))
                .findFirst().orElse(deepest.getCause().getCause().getMessage());
    }
```

Reply:
@mhmxs whats your opinion?